verification/cbmc/include/proof_helpers/make_common_data_structures.h

```diff
@@ -19,8 +19,14 @@
 
 #include <stdlib.h>
 
-#define ASSUME_VALID_MEMORY(ptr) ptr = bounded_malloc(sizeof(*(ptr)))
-#define ASSUME_VALID_MEMORY_COUNT(ptr, count) ptr = bounded_malloc(sizeof(*(ptr)) * (count))
+#define ASSUME_VALID_MEMORY_COUNT(ptr, count)                                                                          \
+    do {                                                                                                               \
+        ptr = bounded_malloc(sizeof(*(ptr)) * (count));                                                                \
+        __CPROVER_assume(AWS_MEM_IS_WRITABLE(ptr, count));                                                             \
```

Comment:
What is this doing here? Shouldn't we do an explicit notnull check?

Reply:
I replace it with a nullness check.